packages/vite/src/node/utils.ts

```diff
@@ -787,10 +787,10 @@ export function processSrcSetSync(
 }
 
 const cleanSrcSetRE =
-  /(?:url|image|gradient|cross-fade)\([^)]*\)|"([^"]|(?<=\\)")*"|'([^']|(?<=\\)')*'/g
+  /(?:url|image|gradient|cross-fade)\([^)]*\)|"([^"]|(?<=\\)")*"|'([^']|(?<=\\)')*'|data:[/;,\w]+/g
```

Comment:
From the original issue's base64 value: `data:image/avif;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAFUlEQVR42mNk+M9Qz0AEYBxVSF+FAAhKDveksOjmAAAAAElFTkSuQmCC`, this doesn't seem to capture the entire `data:...`, specifically the `+` char breaks it.

Should we capture the entire thing until the first space, e.g. `data:\S+`? There could be different `data:` encodings, and `srcset` is space sensitive, so we know that `data:...` shouldn't contain spaces too.

Reply:
I see. I'll look more into capturing base64 values with RE.

Reply:
Here are complete list of all the mime types.

https://www.iana.org/assignments/media-types/media-types.xhtml#image

In my understanding, these srcset would only accept `image/...`. Something like this,

```js
/data:image\/[\w\.\+\-]+;base64,[\w\+\=\/]+/
```

For security reason, I think it should only take image. Or should we open the possibilities to accept other mime types as well? Like this,

```js
/data:\w+\/[\w.+\-]+;base64,[\w+=/]+/
```

Reply:
Yeah I think we can take only `image` as a start, then see if there's any other requests for it in the future. Having a stricter regex is fine by me too, but I think generally it's also fine if it's looser as the worst case is that some URLs are not processed only.

Reply:
I have loosen the Regex. I agree with you that there won't be a big problem with loosening the Regex outside of bad parsing.